modules/space/rendering/renderablestars.cpp

```diff
@@ -533,100 +458,95 @@ RenderableStars::RenderableStars(const ghoul::Dictionary& dictionary)
     )
     , _fixedColor(FixedColorInfo, glm::vec3(1.f), glm::vec3(0.f), glm::vec3(1.f))
     , _filterOutOfRange(FilterOutOfRangeInfo, false)
-    , _pointSpreadFunctionTexturePath(PsfTextureInfo)
-    , _psfMethodOption(
-        PSFMethodOptionInfo,
-        properties::OptionProperty::DisplayType::Dropdown
-    )
-    , _psfMultiplyOption(
-        SizeCompositionOptionInfo,
-        properties::OptionProperty::DisplayType::Dropdown
-    )
-    , _lumCent(LumPercentInfo, 0.5f, 0.f, 3.f)
-    , _radiusCent(RadiusPercentInfo, 0.5f, 0.f, 3.f)
-    , _brightnessCent(BrightnessPercentInfo, 0.5f, 0.f, 3.f)
+    , _halo{
+        properties::PropertyOwner(HaloOwnerInfo),
+        properties::StringProperty(TextureInfo),
+        properties::FloatProperty(PartialOpacityInfo, 1.f, 0.f, 5.f),
+        properties::FloatProperty(ScaleInfo, 1.f, 0.f, 1.f)
+    }
+    , _glare{
+        properties::PropertyOwner(GlareOwnerInfo),
+        properties::StringProperty(TextureInfo),
+        properties::FloatProperty(PartialOpacityInfo, 1.f, 0.f, 5.f),
+        properties::FloatProperty(ScaleInfo, 1.f, 0.f, 1.f)
+    }
+    , _parameters{
+        properties::PropertyOwner(ParametersOwnerInfo),
+        properties::OptionProperty(
+            SizeCompositionOptionInfo,
+            properties::OptionProperty::DisplayType::Dropdown
+        ),
+        properties::FloatProperty(LumPercentInfo, 0.5f, 0.f, 3.f),
+        properties::FloatProperty(RadiusPercentInfo, 0.5f, 0.f, 3.f),
+        properties::FloatProperty(BrightnessPercentInfo, 0.5f, 0.f, 3.f)
+    }
     , _magnitudeExponent(MagnitudeExponentInfo, 6.2f, 5.f, 8.f)
-    , _spencerPSFParamOwner(SpencerPSFParamOwnerInfo)
-    , _p0Param(P0ParamInfo, 0.384f, 0.f, 1.f)
-    , _p1Param(P1ParamInfo, 0.478f, 0.f, 1.f)
-    , _p2Param(P2ParamInfo, 0.138f, 0.f, 1.f)
-    , _spencerAlphaConst(AlphaConstInfo, 0.02f, 0.000001f, 5.f)
-    , _moffatPSFParamOwner(MoffatPSFParamOwnerInfo)
-    , _FWHMConst(FWHMInfo, 10.4f, 0.f, 100.f)
-    , _moffatBetaConst(BetaInfo, 4.765f, 0.f, 100.f)
-    , _renderingMethodOption(
-        RenderMethodOptionInfo,
-        properties::OptionProperty::DisplayType::Dropdown
-    )
-    , _userProvidedTextureOwner(UserProvidedTextureOptionInfo)
-    , _parametersOwner(ParametersOwnerOptionInfo)
-    , _moffatMethodOwner(MoffatMethodOptionInfo)
     , _fadeInDistances(
         FadeInDistancesInfo,
         glm::vec2(0.f),
         glm::vec2(0.f),
         glm::vec2(100.f)
     )
-    , _disableFadeInDistance(DisableFadeInInfo, true)
+    , _enableFadeInDistance(EnableFadeInInfo, false)
 {
-    using File = ghoul::filesystem::File;
-
     const Parameters p = codegen::bake<Parameters>(dictionary);
 
     addProperty(Fadeable::_opacity);
 
-    _dataMapping.bvColor = p.dataMapping.bv.value_or("");
+    _speckFile = p.speckFile.string();
+    _speckFile.onChange([this]() { _speckFileIsDirty = true; });
+    addProperty(_speckFile);
+
+
```

Comment:
```suggestion
```

Reply:
The double linebreak here (and further down) is on purpose to group the different "groups" of properties visually together. Otherwise it was becoming too much of a mess to quickly parse